src/services/lease-service/priority-list-service.ts

```diff
@@ -0,0 +1,178 @@
+import { getWaitingList } from './adapters/xpand/xpand-soap-adapter'
+import {
+  getContactByContactCode,
+  getResidentialAreaByRentalPropertyId,
+  getLeasesForContactCode,
+} from './adapters/xpand/tenant-lease-adapter'
+import app from '../../app'
+
+const getDetailedApplicantInformation = async (applicant: Applicant) => {
+  try {
+    const applicantFromXpand = await getContactByContactCode(
+      applicant.contactCode,
+      'false'
+    )
+
+    if (!applicantFromXpand) {
+      throw new Error(
+        `Applicant ${applicant.contactCode} not found in contact query`
+      )
+    }
+
+    const applicantWaitingList = await getWaitingList(
+      applicantFromXpand.nationalRegistrationNumber
+    )
+    const waitingListForInternalParkingSpace =
+      parseWaitingListForInternalParkingSpace(applicantWaitingList)
+
+    if (!waitingListForInternalParkingSpace) {
+      throw new Error(
+        `Waiting list for internal parking space not found for applicant ${applicant.contactCode}`
+      )
+    }
+
+    const leases = await getLeasesForContactCode(
+      applicant.contactCode,
+      'true', //this filter does not consider upcoming leases
+      undefined //do not include contacts
+    )
+
+    if (!leases) {
+      throw new Error(`Leases not found for applicant ${applicant.contactCode}`)
+    }
+
+    let activeAndUpcomingLeases: Lease[] = leases.filter(
+      isLeaseActiveOrUpcoming
+    )
+
+    for (const lease of activeAndUpcomingLeases) {
+      lease.residentalArea = await getResidentialAreaByRentalPropertyId(
+        lease.rentalPropertyId
+      )
+    }
+
+    let housingContracts = parseLeasesForHousingContracts(
+      activeAndUpcomingLeases
+    )
+    if (!housingContracts) {
+      throw new Error(
+        `Housing contracts not found for applicant ${applicant.contactCode}`
+      )
+    }
+
+    const parkingSpaces = parseLeasesForParkingSpaces(activeAndUpcomingLeases)
+
+    //todo: define the proper interface type to return
+    return {
+      ...applicant,
+      queuePoints: waitingListForInternalParkingSpace.queuePoints,
+      address: applicantFromXpand.address,
+      currentHousingContract: housingContracts[0],
+      upcomingHousingContract: housingContracts[1],
+      parkingSpaceContracts: parkingSpaces,
+    }
+  } catch (error) {
+    console.error('Error in getDetailedApplicantInformation:', error)
+    throw error // Re-throw the error to propagate it upwards
+  }
+}
+
+//helper function to filter all non-terminated and all still active contracts with a last debit date
+const isLeaseActiveOrUpcoming = (lease: Lease): boolean => {
+  const currentDate = new Date()
+  const terminationDate = lease.terminationDate
+    ? new Date(lease.terminationDate)
+    : null
+
+  const lastDebitDate = lease.lastDebitDate
+    ? new Date(lease.lastDebitDate)
+    : null
+
+  return (
+    (!lastDebitDate || currentDate < lastDebitDate) &&
+    (!terminationDate || currentDate < terminationDate)
```

Comment:
Får man be om deskriptiva variabler för dessa två case? Har lite svårt att fatta vad de gör. (vet att kommentaren säger det men tycker ändå det kan styckas upp i två variabler t.ex ... :)